core_lib/src/interface/editor.cpp

```diff
@@ -1078,6 +1079,58 @@ void Editor::swapLayers(int i, int j)
     mScribbleArea->updateAllFrames();
 }
 
+Status::StatusInt Editor::pegBarAlignment(QStringList layers)
+{
+    Status::StatusInt retLeft;
+    Status::StatusInt retRight;
+
+    LayerBitmap* layerbitmap = static_cast<LayerBitmap*>(mLayerManager->currentLayer());
+    BitmapImage* img = layerbitmap->getBitmapImageAtFrame(currentFrame());
+    QRectF rect = select()->mySelectionRect();
+    retLeft = img->findLeft(rect, 121);
+    retRight = img->findTop(rect, 121);
+    if (retLeft.errorcode == Status::FAIL || retRight.errorcode == Status::FAIL)
+    {
+        retLeft.errorcode = Status::FAIL;
+        return retLeft;
+    }
+    int peg_x = retLeft.value;
+    int peg_y = retRight.value;
+
+    // move other layers
+    for (int i = 0; i < layers.count(); i++)
+    {
+        layerbitmap = static_cast<LayerBitmap*>(mLayerManager->findLayerByName(layers.at(i)));
+        for (int k = layerbitmap->firstKeyFramePosition(); k <= layerbitmap->getMaxKeyFramePosition(); k++)
+        {
+            if (layerbitmap->keyExists(k))
+            {
+                img = layerbitmap->getBitmapImageAtFrame(k);
+                retLeft = img->findLeft(rect, 121);
+                if (retLeft.errorcode == Status::FAIL)
+                {
+                    QMessageBox::information(nullptr, nullptr,
```

Comment:
I understand what you want to do here but Editor and Core in general should not contain GUI feedback because in case a test is made, the program will get stuck if something like a qmessagebox is triggered. Although we don't have many tests for now, it should always be possible to run a method when implementing in core, without triggering GUI feedback.

I think it would be a good idea to have a general error or warning popup that can be emitted from Core and accepted in Mainwindow2. I'll make a PR on your branch.